File: frontend-application/src/app/model/Category.ts
import {Status} from "./Status";

export class Category {
  id: string;
  name: string;
  description: string;
  userId: string;
  status: Status;
  color: string;
  countOfnotes: number;

  constructor(id: string, name: string, description: string, userId: string,
              status: Status, color: string, countOfnotes: number) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.userId = userId;
    this.status = status;
    this.color = color;
    this.countOfnotes = countOfnotes;
  }
}

File: backend-application/src/main/java/com/localnotes/entity/Category.java
package com.localnotes.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.List;

@Entity
@Table(name = "categories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Category extends BaseEntity {

    @Column(name = "cat_name")
    private String name;

    @Column(name = "user_id")
    private String userId;

    @Column(name = "description")
    private String description;

    @Column(name = "color")
    private String color;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "category", fetch = FetchType.LAZY)
    private List<Note> noteList;
}
